SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1155,14 +1155,45 @@ - (void)handleSendToMyself:(TSOutgoingMessage *)outgoingMessage
             [TSContactThread getOrCreateThreadWithContactId:contactId transaction:transaction];
         [cThread saveWithTransaction:transaction];
 
+        // We need to clone any attachments for message sent to self; otherwise deleting
+        // the incoming or outgoing copy of the message will break the other.
+        NSMutableArray<NSString *> *attachmentIds = [NSMutableArray new];
+        for (NSString *attachmentId in outgoingMessage.attachmentIds) {
+            TSAttachmentStream *_Nullable outgoingAttachment =
+                [TSAttachmentStream fetchObjectWithUniqueID:attachmentId];
```

Comment:
let's use the existing transaction param rather than opening a new one.

Reply:
Done, good one.